File: test-suite/src/test/java/io/micronaut/xml/jackson/docs/BookSaved.java
package io.micronaut.xml.jackson.docs;

import tools.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import tools.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
import io.micronaut.core.annotation.Introspected;
import org.jspecify.annotations.NonNull;

@JacksonXmlRootElement(localName = "book")
@Introspected
public class BookSaved {

    @NonNull
    private final String name;

    @NonNull
    @JacksonXmlProperty(isAttribute = true)
    private final String isbn;

    public BookSaved(String name, String isbn) {
        this.name = name;
        this.isbn = isbn;
    }

    @NonNull
    public String getName() { return name; }

    @NonNull
    public String getIsbn() { return isbn; }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/xml/jackson/docs/BookSaved.kt
package io.micronaut.xml.jackson.docs;

import tools.jackson.dataformat.xml.annotation.JacksonXmlProperty
import io.micronaut.core.annotation.Introspected
import tools.jackson.dataformat.xml.annotation.JacksonXmlRootElement

@JacksonXmlRootElement(localName = "book")
@Introspected
data class BookSaved(val name: String, @field:JacksonXmlProperty(isAttribute = true) val isbn: String)
